### PR TITLE
fix(forms): Enumeration widget : add componentWillReceiveProps in or…

### PR DESCRIPTION
**Pull request by Sylvain-Talend**
…der to refresh component state when its props changes

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Component state not updated when its props changes

**What is the new behavior?**
Component state  updated when its props changes

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 1]
```diff
@@ -77,6 +77,10 @@ class EnumerationWidget extends React.Component {
 		};
 	}
 
+	componentWillReceiveProps(nextProps) {
+		this.setState({ ...this.state, items: nextProps.formData });
```
Comment (jlame) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: in the constructor the formData object seems to be reworked : 
`items: (props.formData || []).map((item) => ({
				values: item.values,
			})),`

Just wondering why there's a difference in the willReceiveProps ? In that case, the items property directly being assigned the formData object in props.

Comment (Sylvain-Talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: Indeed, the formData is already well formatted and this transformation is unnecessary. I remove it.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js (hunk 1 above)
Comment (jlame) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: ok, this is in fact because the formData is mocked in a js for the moment.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://301.talend.surge.sh)

**Pull request merged**